src/realm/util/serializer.cpp

```diff
@@ -99,6 +100,95 @@ std::string print_value<>(realm::Timestamp t)
     return ss.str();
 }
 
+
+// The variable name must be unique with respect to the already chosen variables at
+// this level of subquery nesting and with respect to the names of the columns in the table.
+// This assumes that columns can start with '$' and that we might one day want to support
+// referencing the parent table columns in the subquery. This is currently disabled by an assertion in the
+// core SubQuery constructor.
+std::string SerialisationState::get_variable_name(ConstTableRef table) {
+    std::string guess_prefix = "$";
+    const char start_char = 'x';
+    char add_char = start_char;
+
+    auto next_guess = [&]() {
+        add_char = (((add_char + 1) - 'a') % 26) + 'a';
+        if (add_char == start_char) {
+            guess_prefix += add_char;
+        }
+    };
+
+    while (true) {
+        std::string guess = guess_prefix + add_char;
+        bool found_duplicate = false;
+        for (size_t i = 0; i < subquery_prefix_list.size(); ++i) {
+            if (guess.compare(subquery_prefix_list[i]) == 0) {
```

Comment:
I think `guess == subquery_prefix_list[i]` is more well known and thus easier to read (I had to look-up if compare() was any special when I reviewed this code)